mpqp/core/instruction/breakpoint.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import annotations
+
+from mpqp.core.instruction import Instruction
+
+
+class Breakpoint(Instruction):
+    """A breakpoint is a special instruction that will display the state of the
+    circuit at the desired state.
+
+    Args:
+        draw_circuit: If ``True`` in addition of displaying the current state
+            vector, it will draw the circuit at this step.
+        enabled: A breakpoint can be kept in the circuit but temporarily
+            disabled using this argument.
+        label: A breakpoint can be given a label for easier traceability.
+    """
+
+    def __init__(
+        self,
+        draw_circuit: bool = False,
+        enabled: bool = True,
+        label: str | None = None,
```

Comment:
Do you think it is useful to keep the info about where this breakpoint is happening in the circuit (like position in the instruction array etc). Because for instance if you use `circuit.breakpoints()` how do you know which one is where exactly ? (you only have how they are ordered I think)

Reply:
indeed, this was a bit annoying to handle: in order to know the position, I had to iterate over the circuit's instructions to get the index. On the other hand, not sure I'd like to store the index of the bp. For instance, if I did
```py
c = QCircuit([Y(0), Breakpoint()])
del c.instructions[0]
```
then I would have the wrong stored index for the breakpoint. The alternative would be to recompute it each tome we modify the instructions, but I don't this this is a better solution

Reply:
Maybe a solution would be to save the reference of the previous gate ? In that case you can quickly re-find where you are putting a breakpoint in the circuti ?

Reply:
After a long discussion, we decided to carry on with the previous implementation, and only change it latter, if we realize it causes problems we couldn't foresee